### PR TITLE
feat/253 - Ledger connection setup page

### LINKED ISSUE #253: Support Ledger devices via Zondax app
We need to support connecting to the Ledger Nano S+ device via the Zondax integration.

See https://github.com/Zondax/ledger-namada

- [x] User can choose to connect their Ledger device
- [x] User is prompted to unlock their Ledger device if needed
- [x] User is directed to a connect to Ledger flow (See: https://support.ledger.com/hc/en-us/articles/4404366864657-Connect-your-Ledger-to-Metamask for an example of what this looks like on Metamask - similar to Keplr)
- [x] User can access accounts on their Ledger device
- [ ] Test transfer transaction signing using keys on their Ledger device
- [ ] Block PR merging until `murisi/separate-signing` is merged to Namada (See https://github.com/anoma/namada/pull/1498)

### PR DESCRIPTION
**Pull request by jurevans**
Begins implementation for #253 

- [x] User can choose to connect Ledger from Setup page
  - _Note_ we can also provide a direct link within the Settings page, same for Create and Import Account, and only show the top-level page during initial setup (will address these in the next PR!):
    - `setup.html#/account-creation/seed-phrase`
    - `setup.html#/import-account`
    - `setup.html#/ledger`
- [x] User must set an `alias`
- [x] User is prompted to allow permissions to connect in browser
- [x] Rework components and routing in Setup pages for consistency
- [x] Upgrade to [v0.0.2](https://www.npmjs.com/package/@zondax/ledger-namada) of the JS package (test this against [v0.0.4](https://github.com/Zondax/ledger-namada/releases/tag/v0.0.4) of NamadaApp)

__TODO__ (separate PR)
- [ ] Ledger account (alias + public key) are tracked in extension for future connections (ledger background service)
- [ ] User is able to see Ledger accounts in extension
- [ ] User is able to see Ledger accounts in interface

This involved a bit of restructuring of the `Setup` app routes to allow for 3 top-level flows with their corresponding sub-routes:
1. `Create Account`
2. `Import Account`
3. `Connect Ledger`

#### Ledger Connection View

Ledger Connect

#### Ledger Connection Success

Ledger Connect Success v0 0 2

#### Ledger Connection Error

Ledger Connect Error

#### Testing

- Install https://github.com/Zondax/ledger-namada/releases/tag/v0.0.4 via `./installer_nanos_plus.sh load` to the Ledger
  - If on macOS, first run: `python3 -m pip install ledgerblue`

### CODE REVIEW

File: apps/extension/src/Setup/AccountCreation/Steps/Password/Password.tsx
Comment (emccorson) [inline apps/extension/src/Setup/AccountCreation/Steps/Password/Password.tsx:73]: Is alias still optional for non-ledger accounts? I can still create an account without an alias, but not sure if that is expected behaviour.

File: apps/extension/src/Setup/AccountCreation/AccountCreation.tsx
Comment (emccorson) [inline apps/extension/src/Setup/AccountCreation/AccountCreation.tsx:27]: The chevron is moving about a bit between pages. I don't think it needs to block the PR - maybe we could add it as a new issue and do it when doing other small UI fixes?

**Review by emccorson — COMMENTED** [review]
I wasn't able to test the ledger parts because I don't have the hardware, but I left a couple of minor comments on the rest and otherwise it looks good!

File: apps/extension/src/Setup/AccountCreation/AccountCreation.tsx
Comment (jurevans) [reply apps/extension/src/Setup/AccountCreation/AccountCreation.tsx:27]: Yeah, it's still a little buggy UI-wise. I can clean that up on this PR. I'm not focusing too much on UI, but it should still be acceptable. I'll be building out the flow in the next PR, and we'll also have to redo it once we get final designs

File: apps/extension/src/Setup/AccountCreation/Steps/Password/Password.tsx
Comment (jurevans) [reply apps/extension/src/Setup/AccountCreation/Steps/Password/Password.tsx:73]: I think we should make it required. The set-up process doesn't require it yet, but elsewhere in the app it is now required.

Comment (jurevans) [reply apps/extension/src/Setup/AccountCreation/Steps/Password/Password.tsx:73]: I've updated the validation to make this a required field!

File: apps/extension/src/Setup/AccountCreation/AccountCreation.tsx
Comment (jurevans) [reply apps/extension/src/Setup/AccountCreation/AccountCreation.tsx:27]: I've decided to rework the components across the `Setup` app to remove all of the duplication that's happening, and should get this issue fixed!

Comment (jurevans) [reply apps/extension/src/Setup/AccountCreation/AccountCreation.tsx:27]: Resolving this as it should all be fixed now

**Review by mateuszjasiuk — APPROVED** [review]
LGTM after reviewing it together.

**Pull request merged**
